File: mcp/src/aieo/src/logger.ts
export interface Logger {
  info(message: string): void;
  error(message: string): void;
  warn?(message: string): void;
  debug?(message: string): void;
  trace?(message: string): void;
  fatal?(message: string): void;
}

export function consoleLogger(name: string): Logger {
  return {
    info(message: string): void {
      console.log(`[${name}] ${message}`);
    },
    error(message: string): void {
      console.error(`[${name}] ${message}`);
    },
    warn(message: string): void {
      console.warn(`[${name}] ${message}`);
    },
    debug(message: string): void {
      console.debug(`[${name}] ${message}`);
    },
    trace(message: string): void {
      console.trace(`[${name}] ${message}`);
    },
    fatal(message: string): void {
      console.error(`==FATAL== [${name}] ${message}`);
    },
  };
}

File: ast/src/testing/vanila/go_scripts/utils/logger.go
package utils

import (
	"fmt"
	"log"
	"os"
	"time"
)

type Logger struct {
	infoLog  *log.Logger
	errorLog *log.Logger
}

func NewLogger() *Logger {
	return &Logger{
		infoLog:  log.New(os.Stdout, "INFO\t", log.Ldate|log.Ltime),
		errorLog: log.New(os.Stderr, "ERROR\t", log.Ldate|log.Ltime|log.Lshortfile),
	}
}

func (l *Logger) Info(format string, v ...interface{}) {
	l.infoLog.Printf(format, v...)
}

func (l *Logger) Error(format string, v ...interface{}) {
	l.errorLog.Printf(format, v...)
}

func (l *Logger) LogRequest(method, path string, duration time.Duration, status int) {
	l.infoLog.Printf("%s %s %d %v", method, path, status, duration)
}

func (l *Logger) Fatal(v ...interface{}) {
	l.errorLog.Output(2, fmt.Sprint(v...))
	os.Exit(1)
}
